### PR TITLE
feat: [CDS-38856]: Add support for package type artifact

### PR DESCRIPTION
**Pull request by abosii-harness**
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>
  
- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>
  
  You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`
  
- Compile: `trigger compile`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

---
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/34523)

### CODE REVIEW

**Review by nassergonzalez — APPROVED** [review]
> Reviewed 36 of 36 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/34523)* status: :shipit: complete! ✓

**Review by vlica-harness — DISMISSED** [review]

File: 120-ng-manager/config.yml [hunk 1]
```diff
@@ -264,7 +264,7 @@ pmsGrpcClientConfig:
   target: localhost:12011
   authority: localhost
 
-shouldConfigureWithPMS: ${SHOULD_CONFIGURE_WITH_PMS:-false}
+shouldConfigureWithPMS: true
```
Comment (vlica-harness) [inline 120-ng-manager/config.yml:267]: this change should be reverted

Comment (abosii-harness) [reply 120-ng-manager/config.yml]: done

**Review by tmacari-harness — CHANGES_REQUESTED** [review]
> Reviewed 3 of 36 files at r1, 39 of 45 files at r2, 4 of 4 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/34523)* status: 2 file(s) to review. 1 change requests. 1. unresolved discussions. 0 of 1 approvals obtained. 

___
*[`930-delegate-tasks/src/main/java/io/harness/delegate/task/azure/appservice/webapp/handler/AbstractSlotDataRequestHandler.java` line 106 at r3](https://reviewable.io/reviews/harness/harness-core/34523#-N8431ABA0HrgxxrKhVz:-N8431ABA0HrgxxrKhW-:b-bvdmv1) ([raw file](https://github.com/harness/harness-core/blob/79589e57fdcef839ed88f9c83d15a972d701ed66/930-delegate-tasks/src/main/java/io/harness/delegate/task/azure/appservice/webapp/handler/AbstractSlotDataRequestHandler.java#L106)):*
> ```Java
>             taskRequest.getStartupCommand() != null ? taskRequest.getStartupCommand().fetchFileContent() : null)
>         .artifactFile(artifactResponse != null ? artifactResponse.getArtifactFile() : null)
>         .artifactType(artifactResponse != null ? artifactResponse.getArtifactType() : ArtifactType.ZIP)
> ```

why do we set here ZIP if artifactResponse is null?
___
*[`950-delegate-tasks-beans/src/main/java/io/harness/delegate/task/azure/appservice/webapp/ng/AzureWebAppInfraDelegateConfig.java` line 35 at r3](https://reviewable.io/reviews/harness/harness-core/34523#-N846G0b7R06HsWquMvH:-N846G0b7R06HsWquMvI:b7rdlzq) ([raw file](https://github.com/harness/harness-core/blob/79589e57fdcef839ed88f9c83d15a972d701ed66/950-delegate-tasks-beans/src/main/java/io/harness/delegate/task/azure/appservice/webapp/ng/AzureWebAppInfraDelegateConfig.java#L35)):*
> ```Java
>   @Expression(ALLOW_SECRETS) private String resourceGroup;
>   @Expression(ALLOW_SECRETS) private String deploymentSlot;
>   @Expression(ALLOW_SECRETS) private String targetSlot;
> ```

not sure if we need targetSlot  for now it seems to be unused.

File: 125-cd-nextgen/src/main/java/io/harness/cdng/azure/webapp/AzureWebAppSlotDeploymentStep.java [hunk 2]
```diff
@@ -281,16 +283,33 @@ private TaskChainResponse queueSlotDeploymentTask(StepElementParameters stepElem
...
         .build();
   }
+
+  private List<String> getCommandUnits(AzureSlotDeploymentPassThroughData passThroughData, boolean fetchFiles) {
+    List<String> units = new ArrayList<>();
+    if (fetchFiles) {
+      units.add(FetchFiles);
+    }
+
+    units.add(SAVE_EXISTING_CONFIGURATIONS);
```
Comment (vlica-harness) [inline 125-cd-nextgen/src/main/java/io/harness/cdng/azure/webapp/AzureWebAppSlotDeploymentStep.java:306]: just asking, shouldn't we have SAVE_EXISTING_CONFIGURATIONS after FETCH_ARTIFACT_FILE?

**Review by abosii-harness — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/34523)* status: 2 file(s) to review. 1 change requests. 4. unresolved discussions. 0 of 1 approvals obtained. 

___
*[`125-cd-nextgen/src/main/java/io/harness/cdng/azure/webapp/AzureWebAppSlotDeploymentStep.java` line 306 at r3](https://reviewable.io/reviews/harness/harness-core/34523#-N7udhrvlFyWZJTMA0JP-r3-306:-N87u-I71t7OfDq-9Ezv:b-j2rls2) ([raw file](https://github.com/harness/harness-core/blob/79589e57fdcef839ed88f9c83d15a972d701ed66/125-cd-nextgen/src/main/java/io/harness/cdng/azure/webapp/AzureWebAppSlotDeploymentStep.java#L306)):*
<details><summary><i>Previously, vlica-harness (Vasile Lica) wrote…</i></summary><blockquote>

just asking,  not sure if this order affects somehow, shouldn't we have SAVE_EXISTING_CONFIGURATIONS after FETCH_ARTIFACT_FILE?
</blockquote></details>

artifact file is downloaded after save existing configuration 
___
*[`930-delegate-tasks/src/main/java/io/harness/delegate/task/azure/appservice/webapp/handler/AbstractSlotDataRequestHandler.java` line 106 at r3](https://reviewable.io/reviews/harness/harness-core/34523#-N8431ABA0HrgxxrKhVz:-N87u5KPAbPOojq_toXV:bg397pd) ([raw file](https://github.com/harness/harness-core/blob/79589e57fdcef839ed88f9c83d15a972d701ed66/930-delegate-tasks/src/main/java/io/harness/delegate/task/azure/appservice/webapp/handler/AbstractSlotDataRequestHandler.java#L106)):*
<details><summary><i>Previously, tmacari-harness (Tudor Macari) wrote…</i></summary><blockquote>

why do we set here ZIP if artifactResponse is null?
</blockquote></details>

It's an enum, not usual for someone to check if enum is null. Just to be safe here
___
*[`950-delegate-tasks-beans/src/main/java/io/harness/delegate/task/azure/appservice/webapp/ng/AzureWebAppInfraDelegateConfig.java` line 35 at r3](https://reviewable.io/reviews/harness/harness-core/34523#-N846G0b7R06HsWquMvH:-N87uEfw2Il5jB6nAFjR:ba3dfob) ([raw file](https://github.com/harness/harness-core/blob/79589e57fdcef839ed88f9c83d15a972d701ed66/950-delegate-tasks-beans/src/main/java/io/harness/delegate/task/azure/appservice/webapp/ng/AzureWebAppInfraDelegateConfig.java#L35)):*
<details><summary><i>Previously, tmacari-harness (Tudor Macari) wrote…</i></summary><blockquote>

not sure if we need targetSlot  for now it seems to be unused.
</blockquote></details>

Removed

**Review by tmacari-harness — APPROVED** [review]
> Reviewed 1 of 45 files at r2, 1 of 1 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/34523)* status: 2. unresolved discussions.

**Review by vlica-harness — COMMENTED** [review]
> Reviewed 3 of 36 files at r1, 30 of 45 files at r2, 4 of 4 files at r3, 1 of 1 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/34523)* status: 2. unresolved discussions.

**Review by vlica-harness — DISMISSED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/34523)* status: :shipit: complete! ✓

**Review by vlica-harness — APPROVED** [review]

**Review by tmacari-harness — APPROVED** [review]
> Reviewed 1 of 1 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/34523)* status: :shipit: complete! ✓

**Review by anilchowdhury-harness — APPROVED** [review]
:lgtm:

> Reviewed 3 of 36 files at r1, 39 of 45 files at r2, 4 of 4 files at r3, 1 of 1 files at r4, 1 of 1 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/34523#-:-N8NNi-o1yrcNKEssuTC:bnfp4nl)* status: :shipit: complete! ✓

**Comment by abosii-harness** [comment]
trigger codebasehash

**Pull request merged**